### ISSUE TITLE
OS X App Freezes/Crashes

### ISSUE THREAD

**Issue by blakfeld** [turtl/desktop#43]
I don't yet have an idea of cause, and I'm unsure where to find more useful diagnostic information, but I've had it happen 3 times this morning.

Basically I'll be editing a note, and walk away for a bit. When I come back the application will be totally unresponsive and frozen. On top of that whatever I was editing has been lost.

I'm running an Early 2015 Macbook Pro with El Capitan 10.11.5

**Comment by orthecreedence** [comment]
Can you hit `f12` and leave the "Console" tab open? I'd like to know if any errors appears when it freezes. Thanks!

**Comment by blakfeld** [comment]
I've got the console open, but naturally now it's working just fine :)

I'll keep it open and watch it for a bit, if it doesn't happen again I'll close this issue.

**Comment by blakfeld** [comment]
Happened again :( This time while in the middle of typing a note. Nothing in the console that wasn't there from the start.

![image](https://cloud.githubusercontent.com/assets/1133823/16672027/dd460006-4468-11e6-9e43-3ab901035927.png)

**Comment by orthecreedence** [comment]
Damn it, of course there's nothing in the console. Typical. 

There has to be some kind of process blocking in the background, but I can't think what it would be. My guess is something to do with the sync system. Sorry to do this, but can you do the same test again, but with the Network tab open?

**Comment by blakfeld** [comment]
Yeah totally, Whatever I can do to get more data! I'm a fan of the app, and I'm looking forward to using it to replace nvAlt/Evernote

**Comment by blakfeld** [comment]
Here's a screenshot of the Network Tab. It seems to be fine if I leave it alone (I left it running all weekend), but when there's actually new content it starts to get squirrelly.
 
![screenshot 2016-07-11 10 30 52](https://cloud.githubusercontent.com/assets/1133823/16736472/ad51fc8c-4752-11e6-9096-9fedd062d193.png)

This is the full output of "copy as cURL" minus auth headers and client_id:

```
curl 'https://api.turtl.it/v2/sync?sync_id=5783ad302b137571090018ec&client_id=$CLIENT_ID&immediate=0' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.79 Safari/537.36' -H 'X-Turtl-Client: desktop-0.6.4'
```

**Comment by crasch** [comment]
FYI, I'm experiencing the same behavior on a MacBook Pro (13-inch, Mid-2012) running Mac OS X Yosemite (10.10.5).

**Comment by blakfeld** [comment]
Yeah, this has lead to me completely abandoning turtl. Which is a shame, because its a good tool.

**Comment by oskapt** [comment]
since @blakfeld abandoned turtl, i'm willing to pick this up to assist with debugging. i'm using the desktop app v0.6.4 (2704.79) and every so often it freezes. it's not a big deal for me to kill it and re-open it, but it would be better if it didn't freeze.

on the server side i'm running turtl in a Docker container behind HAProxy. let me know how i can help.
